packages/wp-story-editor/src/components/mediaUpload/index.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { isValidElement } from '@web-stories-wp/react';
+
+/**
+ * Internal dependencies
+ */
+import { useMediaPicker } from './useMediaPicker';
+
+function MediaUpload({
+  render,
+  onSelect,
+  onClose,
+  onSelectErrorMessage,
+  allowedMimeTypes,
+  setIsPermissionDialogOpen,
+  title,
+  buttonInsertText,
+  cropParams,
+  multiple,
+}) {
+  const open = useMediaPicker({
+    onSelect,
+    onSelectErrorMessage,
+    onClose,
+    type: allowedMimeTypes,
+    title,
+    buttonInsertText,
+    cropParams,
+    multiple,
+    onPermissionError: () => setIsPermissionDialogOpen(true),
+  });
+
+  if (!render || !isValidElement(render())) {
+    return null;
+  }
```

Comment:
Don't think we need those safe guards. I'd rather add prop types here for `MediaUpload` to enforce `render` being a function.